appengine/src/main/java/com/google/googleinterns/gscribe/services/impl/SpreadsheetServiceImpl.java

```diff
@@ -0,0 +1,271 @@
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.googleinterns.gscribe.services.impl;
+
+import com.google.api.client.auth.oauth2.BearerToken;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.sheets.v4.Sheets;
+import com.google.api.services.sheets.v4.model.*;
+import com.google.googleinterns.gscribe.dao.UserTokenDao;
+import com.google.googleinterns.gscribe.models.*;
+import com.google.googleinterns.gscribe.resources.io.exception.InvalidDatabaseDataException;
+import com.google.googleinterns.gscribe.resources.io.exception.InvalidRequestException;
+import com.google.googleinterns.gscribe.services.SpreadsheetService;
+import com.google.googleinterns.gscribe.services.TokenService;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SpreadsheetServiceImpl implements SpreadsheetService {
+
+    private final TokenService tokenService;
+    private final UserTokenDao userTokenDao;
+
+    public SpreadsheetServiceImpl(TokenService tokenService, UserTokenDao userTokenDao) {
+        this.tokenService = tokenService;
+        this.userTokenDao = userTokenDao;
+    }
+
+    /**
+     * Called when responses sheet needs to be setup for the newly created exam
+     * Fetch all the sheets metadata in the spreadsheet and check if sheet with name sheetName already exists
+     * If sheet already exists then clear the sheet
+     * If sheet does not exist then create new sheet with name sheetName in the spreadsheet
+     *
+     * @param service       ( sheet service object )
+     * @param spreadsheetId ( spreadsheet id of spreadsheet where question paper lies )
+     * @param sheetName     ( sheet name of sheet which is to be created or cleared )
+     * @throws IOException ( if any error occurs regarding access of spreadsheet )
+     */
+    private void sheetSetup(Sheets service, String spreadsheetId, String sheetName) throws IOException {
+
+        Sheets.Spreadsheets.Get spreadsheetService = service.spreadsheets().get(spreadsheetId);
+        Spreadsheet spreadsheet = spreadsheetService.execute();
+        List<Sheet> sheets = spreadsheet.getSheets();
+        boolean sheetAlreadyMade = false;
+        for (Sheet sheet : sheets) {
```

Comment:
Separate it out into multiple functions.

Reply:
Broke this into two separate functions.